core/src/Threads/Kokkos_ThreadsExec.cpp

```diff
@@ -834,37 +835,7 @@ const char *Threads::name() { return "Threads"; }
 namespace Impl {
 
 int g_threads_space_factory_initialized =
-    initialize_space_factory<ThreadsSpaceInitializer>("050_Threads");
-
-void ThreadsSpaceInitializer::initialize(
-    const InitializationSettings &settings) {
-  Kokkos::Threads::impl_initialize(
-      settings.has_num_threads() ? settings.get_num_threads() : -1);
-}
-
-void ThreadsSpaceInitializer::finalize(const bool all_spaces) {
-  if (std::is_same<Kokkos::Threads, Kokkos::DefaultExecutionSpace>::value ||
-      std::is_same<Kokkos::Threads,
-                   Kokkos::HostSpace::execution_space>::value ||
-      all_spaces) {
-    if (Kokkos::Threads::impl_is_initialized())
-      Kokkos::Threads::impl_finalize();
-  }
-}
-
-void ThreadsSpaceInitializer::fence(const std::string &name) {
-  Kokkos::Threads::impl_static_fence(name);
-}
-
-void ThreadsSpaceInitializer::print_configuration(std::ostream &msg,
-                                                  const bool detail) {
-  msg << "Host Parallel Execution Space:" << std::endl;
-  msg << "  KOKKOS_ENABLE_THREADS: ";
-  msg << "yes" << std::endl;
-
-  msg << "\nThreads Runtime Configuration:" << std::endl;
-  Kokkos::Threads::print_configuration(msg, detail);
-}
+    initialize_space_factory<Threads>("050_Threads");
```

Comment:
what is 050_?

Reply:
This is a good question.  Currently, this string is used to enforce an order in the registered backends.  These are stored in a `std::map`.  I am not sure why it was chosen over just `std::priority_queue` and giving say priority 3 to device backends, 2 to host parallel backends, and 1 to the serial backend.
I did not want to get into this too much in this PR though,